packages/resolver/src/adapters/handle-api.ts

```diff
@@ -0,0 +1,106 @@
+import {Api, Resolver} from '@yoroi/types'
+import {fetchData, FetchData, handleApiError, isLeft} from '@yoroi/common'
+import {z} from 'zod'
+import {AxiosRequestConfig} from 'axios'
+
+import {handleZodErrors} from './zod-errors'
+
+const initialDeps = {request: fetchData} as const
+
+export const handleApiGetCryptoAddress = ({
+  request,
+}: {request: FetchData} = initialDeps) => {
+  return async (
+    resolve: Resolver.Receiver['resolve'],
+    fetcherConfig?: AxiosRequestConfig,
+  ): Promise<string> => {
+    if (!isAdaHandleDomain(resolve)) throw new Resolver.Errors.InvalidDomain()
+
+    const sanitizedDomain = resolve.replace(/^\$/, '')
+    const config = {
+      url: `${handleApiConfig.mainnet.getCryptoAddress}${sanitizedDomain}`,
+    } as const
+
+    try {
+      const response = await request<HandleApiGetCryptoAddressResponse>(
+        config,
+        fetcherConfig,
+      )
+
+      if (isLeft(response)) {
+        handleApiError(response.error)
+      } else {
+        const parsedResponse = HandleApiResponseSchema.parse(
+          response.value.data,
+        )
+        return parsedResponse.resolved_addresses.ada
+      }
+    } catch (error: unknown) {
+      return handleHandleApiError(error)
+    }
+  }
+}
+
+// https://github.com/koralabs/handles-public-api/blob/master/src/models/view/handle.view.model.ts
```

Comment:
i think this should refer to the current commit

Reply:
that makes sense, can you add as a suggestion ?

Reply:
https://github.com/koralabs/handles-public-api/blob/fd9d9f2cf3143e317a780b81b22869a755ab6af8/src/models/view/handle.view.model.ts